### ISSUE TITLE
Minimum area bounding rectangle

### ISSUE THREAD

**Issue by Sklavit** [Toblerity/Shapely#354]
Please, add computation of the minimum area bounding rectangle.
Possible implementations can be found at:
http://stackoverflow.com/questions/13542855/python-help-to-implement-an-algorithm-to-find-the-minimum-area-rectangle-for-gi?rq=1
https://github.com/dbworth/minimum-area-bounding-rectangle/tree/master/python
https://geidav.wordpress.com/2014/01/23/computing-oriented-minimum-bounding-boxes-in-2d/

**Comment by snorfalorpagus** [comment]
The algorithm given by whuber in this GIS.SE answer looks nice to me: http://gis.stackexchange.com/a/22934/12420. Only 12 lines and no trig functions. With the existing convex hull method provided by GEOS this should be possible to implement.

**Comment by georgeouzou** [comment]
I think computing the mbr is already accomplished by geometry.envelope by geos

**Comment by snorfalorpagus** [comment]
@georgeouzou The geometry.envelope method only calculates the MBR with sides parallel to the coordinate axis. These methods find the smallest rectangle, with rotation.

**Comment by snorfalorpagus** [comment]
It looks like GEOS already has an algorithm for this. It should be trivial to implement.

```
extern GEOSGeometry GEOS_DLL *GEOSMinimumRotatedRectangle(const GEOSGeometry* g);
```

https://github.com/libgeos/libgeos/blob/a291393068a45150313140d57972d9c914e21f21/capi/geos_c.h.in#L1475

I'll do this when I find a little time.

**Comment by georgeouzou** [comment]
@Sklavit  I just submitted PR #361 that adds the new functionality to compute the rotated minimum rectangle. I thought this might be of interest to you.

**Comment by sgillies** [comment]
Merged #361

**Comment by BebeSparkelSparkel** [comment]
Hey I think this is great to add. Unfortunately I already wrote the code for [Minimum Bounding Box](https://github.com/BebeSparkelSparkel/MinimumBoundingBox) but I also use the [Smallest Enclosing Circle](https://github.com/BebeSparkelSparkel/SmallestEnclosingCircle) and it would be nice if that could be added.

Lastly, I noticed that the bounds method does not cache the result and if called a lot it slows down the execution time a lot. It would be nice if the bounds method, rotated minimum rectangle, and if added smallest enclosing circle could have their results cached to save execution time.
